class/User.js

```diff
@@ -402,17 +403,23 @@ export class User {
     }
 
     try {
-      let txs = await this._bitcoindrpc.request('listtransactions', ['*', 100500, 0, true]);
-      // now, compacting response a bit
-      let ret = { result: [] };
-      for (const tx of txs.result) {
-        ret.result.push({
-          category: tx.category,
-          amount: tx.amount,
-          confirmations: tx.confirmations,
-          address: tx.address,
-          time: tx.time,
-        });
+      if (config.bitcoind) {
+        let txs = await this._bitcoindrpc.request('listtransactions', ['*', 100500, 0, true]);
+        // now, compacting response a bit
+        let ret = { result: [] };
+        for (const tx of txs.result) {
+          ret.result.push({
+            category: tx.category,
+            amount: tx.amount,
+            confirmations: tx.confirmations,
+            address: tx.address,
+            time: tx.time,
+          });
+        }
+      } else {
+        let ret = { result: [] };
+        let txs = await this._getChainTransactions();
+        ret.result.push(...txs);
       }
       _listtransactions_cache = JSON.stringify(ret);
```

Comment:
looks like `ret` definition has to be moved one level up
```
2021-02-24T12:44:36.245699+00:00 app[web.1]: listtransactions error: ReferenceError: ret is not defined
2021-02-24T12:44:36.245708+00:00 app[web.1]:     at User._listtransactions (/app/class/User.js:424:48)
2021-02-24T12:44:36.245709+00:00 app[web.1]:     at runMicrotasks (<anonymous>)
2021-02-24T12:44:36.245709+00:00 app[web.1]:     at processTicksAndRejections (internal/process/task_queues.js:93:5)
2021-02-24T12:44:36.245710+00:00 app[web.1]:     at User.getTxs (/app/class/User.js:336:15)
2021-02-24T12:44:36.245710+00:00 app[web.1]:     at /app/controllers/api.js:397:15
```

Reply:
Sorry, I'll fix it.